massalabs/coin-vester
# Reinit frontend project

- vite app, react-typescript
- code formatting: prettier
- linter: eslint
- precommit hook: lintstaged with husky
- CI for code formatting and linter
- component library: massalabs ui kit
- CSS framework tailwindcss
- color theme
- internationalization: en_US
- remake HomePage
- remake SendPage
- hooks for SC communication

QA:
If not deployed, do those steps first:
1. Make sure you have Node and NPM installed
2. Clone the repository
3. Using a terminal, go to the repository
4. go to the front end dir: `cd coin-vester-front`
5. install dependencies: `npm i`
6. start front end: `npm run dev`
7. You should now be able to access front end using your terminal on http://localhost:5173/

The project is divided in 3 pages:
- http://localhost:5173/ The "Home Page" (or "Claim Page") which allows users to claim their vesting
- http://localhost:5173/send The "Send Page" which will be mainly used internally to send vesting sessions
- http://localhost:5173/legacy The "Legacy Page" which contains the old version of the front end, just in case we need it

1. Choose the wallet you want to use, Bearby or Massa Wallet
2. Open your wallet and make sure to be on buildnet
  a. On MW: Open https://station.massa and change the network to buildnet using the select on the top right
  b. On Bearby: Open Bearby, Click on the settings button at the bottom right, click on "Network", change network to buildnet
3. [Open the Send Page](http://localhost:5173/send)
4. Connect your wallet
5. Choose an account with coins you want to use to send vesting (ideally, one with coins, if needed, as @thomas-senechal for coins on buildnet)
6. Choose the account you want to send vesting to and copy the address
7. Try to break the form by putting wrong values. If something seems wrong, ping @thomas-senechal .
8. Fill the form for real using the address account you previously copied
9. Once the form is filled, click on the "Send vesting" button at the bottom and confirm
10. [Open the Claim Page](http://localhost:5173/)
11. Connect your wallet
12. Choose the account you sent vesting to
13. You should see the vesting session your created
14. Click on more infos to check if the informations match the ones you sent earlier
15. Try to break the "Amount to claim" input
16. Put a valid Amount to claim and press the "Claim" button
17. Everything should work fine
